src/System.Reflection.Metadata/tests/TestUtilities/LoaderUtilities.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.IO;
+using System.Reflection.PortableExecutable;
+using Xunit;
+
+namespace System.Reflection.Metadata.Tests
+{
+    internal unsafe static class LoaderUtilities
+    {
+        public static void LoadPEAndValidate(byte[] peImage, Action<PEReader> validator)
```

Comment:
> I assume the loader logic is implemented in CoreCLR

It is here: https://github.com/dotnet/coreclr/blob/b49116c0f940760c49b2af9b7593e38a11921694/src/pal/src/map/map.cpp#L2283 but it is designed to load native images only. It will fail for regular IL only images.